src/crawlee/browsers/_playwright_browser_plugin.py

```diff
@@ -70,6 +70,7 @@ def __init__(
         self._browser_launch_options = default_launch_browser_options | (browser_launch_options or {})
         self._browser_new_context_options = browser_new_context_options or {}
         self._max_open_pages_per_browser = max_open_pages_per_browser
+        self._use_incognito_pages: bool = self._browser_launch_options.pop('use_incognito_pages', False)
```

Comment:
The `use_incognito_pages` option should not be included in the `browser_launch_options`. This parameter is intended for browser launch options supported by Playwright ([docs](https://playwright.dev/python/docs/api/class-browsertype#browser-type-launch)).

The `use_incognito_pages` is our custom flag used for internal logic in our code, not by Playwright. Mixing it with `browser_launch_options` does not seem correct. Instead, it should be a dedicated argument.

And it should be probably exposed in the same places as `browser_launch_options`, specifically: `PlaywrightCrawler.__init__`, `BrowserPool.with_default_plugin`, `PlaywrightBrowserPlugin.__init__`.